nixos/modules/services/x11/xidlehook.nix

```diff
@@ -0,0 +1,225 @@
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      notifier = mkOption {
+        default = null;
+        example = "${pkgs.libnotify}/bin/notify-send \"Locking in ${cfg.notify} seconds\"";
+        type = types.nullOr types.str;
+
+        description = ''
+          Notification script to be used to warn about the pending autolock.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      killer = mkOption {
+        default = null; # default according to `man xautolock` is none
+        example = "${pkgs.systemd}/bin/systemctl suspend";
+        type = types.nullOr types.str;
+
+        description = ''
+          The script to use when nothing has happend for as long as <option>killtime</option>
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      killtime = mkOption {
+        default = 20; # default according to `man xautolock`
+        type = types.int;
+
+        description = ''
+          Minutes xidlehook waits until it executes the script specified in <option>killer</option>
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+    };
+  };
+
+  config = {
+    services.xserver.xidlehook.timers = mkDefault (
+      (optional cfg.enableNotifier {
+        duration = cfg.time * 60 - cfg.notify;
+        command = cfg.notifier;
+      }) ++ [{
+        duration = if cfg.enableNotifier then cfg.notify else cfg.time * 60;
+        command = cfg.locker;
+      }] ++ (optional (cfg.killer != null) {
+        duration = (cfg.killtime - cfg.time) * 60;
+        command = cfg.killer;
+      }));
+
+    environment.systemPackages = optional cfg.enable pkgs.xidlehook;
+
+    systemd.user.services.xidlehook = mkIf cfg.enable {
+      description = "xidlehook service";
+      wantedBy = [ "graphical-session.target" ];
+      partOf = [ "graphical-session.target" ];
+      serviceConfig = {
+        ExecStart = strings.concatStringsSep " "
+          ([ "${pkgs.xidlehook}/bin/xidlehook" ] ++ cfg.extraOptions ++ flatten
+            (map (timer: [
+              "--timer"
+              (toString timer.duration)
+              (escapeShellArg timer.command)
+              (escapeShellArg (timer.canceller or ""))
+            ]) cfg.timers));
+        Restart = "always";
+      };
+    };
+
+    assertions = optionals cfg.enable (map (timer: {
+      assertion = builtins.substring 0 1 timer.command == "/";
+      message = "Please specify canonical paths for `services.xserver.xidlehook.timers` commands";
+    }) cfg.timers) ++ (map (timer: {
+      assertion = (timer.canceller or "") != "" -> builtins.substring 0 1 timer.command == "/";
```

Comment:
Shouldn't this be `timer.canceller`?

And this shouldn't be necessary if you set the type of `canceller` to `nullOr path`. Same for the other options, specifying the `path` type automatically ensures an initial `/`

Reply:
> Shouldn't this be `timer.canceller`?

Yes!  Fixed.  Thanks!

> And this shouldn't be necessary if you set the type of `canceller` to `nullOr path`. Same for the other options, specifying the `path` type automatically ensures an initial `/`

It's not a path, though.  It's a command --  it's a path up to the first space, and then not.  In addition to being misleading as documentation, this would break if the path type later gained additional validation like checking path name component length (chars between /s)  or normalizations like replacing /./ with /